assets/wizards/advertising/components/ad-unit-size-control/index.js

```diff
@@ -29,50 +34,73 @@ export const DEFAULT_SIZES = [
 	[ 320, 50 ],
 	[ 320, 100 ],
 	[ 120, 60 ],
+	'fluid',
 ];
 
 /**
  * Ad Unit Size Control.
  */
-const AdUnitSizeControl = ( { value, onChange } ) => {
+const AdUnitSizeControl = ( { value, selectedOptions, onChange } ) => {
 	const [ width, height ] = value;
 	const [ isCustom, setIsCustom ] = useState( false );
-	const sizeIndex = DEFAULT_SIZES.findIndex( size => size[ 0 ] === width && size[ 1 ] === height );
+	const options = DEFAULT_SIZES.filter(
+		size =>
+			JSON.stringify( value ) === JSON.stringify( size ) ||
+			! selectedOptions.find(
+				selectedOption => JSON.stringify( selectedOption ) === JSON.stringify( size )
+			)
+	);
+	const sizeIndex = isCustom
+		? -1
+		: options.findIndex(
+				size => value === size || ( size[ 0 ] === width && size[ 1 ] === height )
+		  );
```

Comment:
I have a feeling this logic is overly complex, and it is hard to follow. I see a following bug but at first glance it's hard to find the culprit:

1. Start with an ad unit with two sizes: one fixed from the list and fluid
2. Change the fluid one (second size) to custom - observe that the fixed one can't be set to fluid (I expected it to be, so I can have a `[custom, fluid]` sizes list – correct me if that's a wrong assumption)
3. Delete the fixed size (first size) – the second size, which was custom, changes to fluid (I expected it possible to have just one size of custom value)

Reply:
b55ef8f78b48efc453e68c49cd2c21cedfe7121d improves the readability of the `findIndex` operation.

The problem you were seeing is that the `Custom` selection does not initially trigger `onChange()` to the parent component, keeping the previously selected value until a custom width or height is filled. ed3f475101af75df748ab1fcbc3ee4312eb869d8 changes that behaviour.